kythe/go/indexer/emit.go

```diff
@@ -163,9 +163,9 @@ type emitter struct {
 	pi       *PackageInfo
 	sink     Sink
 	opts     *EmitOptions
-	impl     map[impl]bool                        // see checkImplements
+	impl     map[impl]struct{}                    // see checkImplements
 	rmap     map[*ast.File]map[int]metadata.Rules // see applyRules
-	anchored map[ast.Node]bool                    // see writeAnchor
+	anchored map[ast.Node]struct{}                // see writeAnchor
```

Comment:
If you find yourself re-implementing this a lot, you may also find it useful to know that the `stringset` code is actually generated and can be re-generated for other types easily.

```shell
# N.B. This example omits the set transforms like Map, Select, etc.
# Set "transforms": true if you want them.
cat > nodeset.json <<EOF
{
  "package": "nodeset",
  "type": "ast.Node",
  "zero": "nil",
  "less": "if x.Pos() == y.Pos() { return x.End() > y.End() } else { return x.Pos() < y.Pos() }",
  "imports": ["go/ast"],
  "transforms": false
}
EOF
go run bitbucket.org/creachadair/stringset/makeset -config makeset/nodeset.json -output nodeset
```

I haven't wired this up to Bazel specifically, but it could be done pretty easily I think with a genrule. YMMV.